keps/0002-kep-moveengine-hld.md

```diff
@@ -0,0 +1,152 @@
+---
+kep-number: 1
+title: MoveEngine - High Level Design
+authors:
+  - "@vitta"
+owners:
+  - Uma Mukkara
+  - Mayank
+editor: 
+creation-date: 2019-05-16
+last-updated: 2019-05-19
+status: provisional
+see-also:
+  - KEP-None
+replaces:
+  - KEP-None
+superseded-by:
+  - KEP-None
+---
+
+# KubeMove - High Level Design
+
+
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Design acceptance criteria](#design-acceptance-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+From the KubeMove HLD KEP, KubeMoveEngine CRD defines the placeholders for the workflow for a given application. The CR is created by the user by invoking the spec that has the following details
+   - Application to which this CR applies
+   - Target location including the namespace. User can specify multiple target locations, in which case multiple KMovePair CRs get created.
+   - The Dynamic Data Mobilizer or DDM details
+   - schedule time to transfer data at regular interval
+   - Telemetry management
+
+`MoveEngine` controller that is watching this CR will deploy the KubeMove framework(`MoveEngine`, `DataSync` controller) at remote cluster to
+setup the application and transfer the data to remote cluster. This controller creates the datasync CR periodically and invokes the `DataSync`
+controller to initiate the data transfer through DDM. Once DDM completes the data transfer, `MoveEngine` controller updates the remote `DataSync` controller by creating dataSync CR at remote cluster to validate/verify the data transfer, and to perform the restore of data.
+
+## Motivation
+
+This KEP is to get into details of MoveEngine CR
+
+### Goals
+
+This KEP is to get into details of MoveEngine CR
+
+### Non-Goals
+
+## Proposal
+
+`spec` of MoveEngine CR consists of following fields:
```

Comment:
This "Move" operator could use human-in-the-loop model for Approve/Deny. It can be similar to `CertificateSigningRequest` in k8s. https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/#create-a-certificate-signing-request-object-to-send-to-the-kubernetes-api

In that case `MoveRequest` could the more appropriate.